src/foundations/rhi/Texture.zig

```diff
@@ -4,45 +4,68 @@ handle: c.GLuint64 = 0,
 uniform: Uniform = undefined,
 wrap_s: c.GLint = c.GL_CLAMP_TO_EDGE,
 wrap_t: c.GLint = c.GL_CLAMP_TO_EDGE,
+disable_bindless: bool = false,
 
 const Texture = @This();
 
 pub const TextureError = error{
     BindlessHandleCreationFailed,
-    BindlessNotSupported,
 };
 
-pub fn init(image: *assets.Image, program: u32, uniform_name: []const u8) TextureError!Texture {
-    var t: Texture = .{};
-    return t.setup(image, program, uniform_name);
+pub fn frag_shader(tx: ?Texture) Shader.fragment_shader_type {
+    if (tx) |t| if (!t.disable_bindless) return .bindless;
+    return .texture;
 }
 
-pub fn setup(self: *Texture, image: *assets.Image, program: u32, uniform_name: []const u8) TextureError!Texture {
+pub fn init(disable_bindless: bool) TextureError!Texture {
+    var t: Texture = .{
+        .disable_bindless = disable_bindless,
+    };
+
     if (c.glfwExtensionSupported("GL_ARB_bindless_texture") != 1 or c.glfwExtensionSupported("GL_ARB_bindless_texture") != 1) {
-        return TextureError.BindlessNotSupported;
+        t.disable_bindless = true;
     }
+    return t;
+}
 
+pub fn setup(self: *Texture, image: ?*assets.Image, program: u32, uniform_name: []const u8) TextureError!void {
```

Comment:
**suggestion (bug_risk):** Consider more robust error handling in texture setup

The current implementation might silently fail in some error cases. Consider propagating errors or logging them for better debugging and reliability.